### DISCUSSION TITLE
(Question) about video streaming in sciter-js

### DISCUSSION THREAD

**Discussion by martinlombana** [c-smile/sciter-js-sdk#63] [Q&A] [answered]
Is video rendering supported? I am trying to load in scapp.exe the simplest of the video-examples (with no TISSCRIPT) but it does not render the vídeo. Is there anything that must be done differently? ( Checked and the URL is ok and loads in a browser)

**test-video-streaming.htm**

Thanks in advance.

`<html>
    <head>
        <title>Test</title>
        <style>

          video {
            background: gray;
            size: *;
          }

        </style>
        <script type="text/tiscript"></script>
    </head>
    <body>

      Streaming from http://clips.vorwaerts-gmbh.de/VfE_html5.mp4 

      <video src="http://clips.vorwaerts-gmbh.de/VfE_html5.mp4" sizing=contain />
      

    </body>
</html>`

**Comment by MustafaHi** [comment]
It does work, try to install [K-Lite Codec Pack](https://codecguide.com/download_kl.htm)

**Comment by martinlombana** [comment]
There is no way to embed that codec, or use a codec that everyone usually
has? Or a way to embed the codec with the sources, so it does work in all
the installs of a sciter developed app?

I also tried with another normal mp4 file and it didn't work either. My
concern is to make it work in an ubiquitous way. Without external deps, or
with those deps, injected into the installation folder/code.

**Comment by martinlombana** [comment]
For myself & others:
https://sciter.com/forums/topic/embedding-videos-in-html-with-sciter/#post-58383
(I missed this before).

What is "weird", is that local videos are indeed reproduced without the
need for extra codecs (at least in windows). How does that work? Would it
behave the same way in a raspberry pi, locally?

Thanks,

**Comment by c-smile** [comment, marked as answer]
Sciter does not contain any video codecs but it allows to connect external sources of video frames for video rendering.

I know several Sciter based projects are using ffmpeg to playback video in Sciter for example.

Here is demo of custom video frame source:
https://github.com/c-smile/sciter-sdk/blob/master/include/behaviors/behavior_video_generator_full.cpp

You can see it in action using wsciter.exe from Sciter SDK by loading this document in it : https://github.com/c-smile/sciter-sdk/blob/master/samples/video/video-generator-behavior-full-demo.htm

**Comment by martinlombana** [comment]
I see. I will digg into that because I would need ffmpeg to work on raspberry pi / windows. 

I see that with **sciter-js there is no wsciter.exe**, I see it in the regular sdk. What is exactly the difference between wsciter and scapp.exe? And usciter?

And why that example only works with wsciter.exe? 

I tried looking for info in the website/docs, but couldn't find it. Thanks.

**Comment by c-smile** [comment]
wsciter.exe includes that native behavior so it works there.

You can add that behavior to uscterjs application, re-compile and that sample will work with SciterJS too. 

> What is exactly the difference between wsciter and scapp.exe? And usciter?

1. wsciter uses sciter.dll and is Windows only project.
2. usciter uses sciter.dll and is universal - is built for all platforms. 
3. scapp is standalone (a.k.a. portable) sciter executable - it contains sciter linked statically therefore sizeof(scapp) ~ sizeof(sciter.dll).
4. Scapp executable works as it is - does not need any other dlls.
